src/pages/index.tsx

```diff
@@ -16,8 +16,8 @@ export const getServerSideProps = async () => {
   };
 };
 
-const Home = ({ todos }: Props) => {
-  return <TodoTable todos={todos} />;
+const HomePage = ({ todos }: Props) => {
+  return withAuthRedirect({ todos });
 };
 
-export default Home;
+export default HomePage;
```

Comment:
HOC should be used like this `export default withAuthRedirect(HomePage);` not inside the component itself